src/types/IstioObjects.ts

```diff
@@ -139,6 +139,39 @@ export interface PodLogs {
   entries: LogEntry[];
 }
 
+export interface EnvoyProxyDump {
+  configs: any[];
+}
+
+export type EnvoyProxyResourceEntries = BootstrapSummary | ClusterSummary | ListenerSummary | RouteSummary;
```

Comment:
I have a comment here.
These optional types look good on paper but in practice it's hard to check in runtime which type is coming from as there are not strong types.

So I guess if we can put a variable for any of these types and check if defined to see if the EnvoyProxyResourceEntries refers to a BootstrapSummary or a ClusterSummary.

Because this comes from an invocation

```
export const getPodEnvoyProxyResourceEntries = (namespace: string, pod: string, resource: string) => {
  return newRequest<EnvoyProxyResourceEntries>(
    HTTP_VERBS.GET,
    urls.podEnvoyProxyResourceEntries(namespace, pod, resource),
    {},
    {}
  );
};
```

So, perhaps this is missing in my backend PR review but instead to have this type pretty open:

```
type ResourceDump interface {
	Parse(dump *kubernetes.ConfigDump) error
}
```

I'd suggest to have an entry for the one of the "clusters", "routes", "boostrap", "listeners",

```
type ResourceDump interface {
         Clusters *models.Clusters,
         Routes *models.Routes,
         Bootstrap *models.Bootstrap,
         Listeners *models.Listeners,
	Parse(dump *kubernetes.ConfigDump) error
}
```
In that way, the caller would know the returning type just checking if that field is populated, not just relying on the parameter asked, I think.